### ISSUE TITLE
`haddock-style:single-line` breaks comments and causes compilation errors.

### ISSUE THREAD

**Issue by jonathanknowles** [fourmolu/fourmolu#279]
### Is your bug specific to Fourmolu?
- [x] Yes, I've verified that Ormolu does not have this bug

### Which versions did you use?
```
$ ormolu --version
ormolu 0.5.0.1 UNKNOWN UNKNOWN
using ghc-lib-parser 9.2.5.20221107

$ fourmolu --version
fourmolu 0.9.0.0 UNKNOWN UNKNOWN
using ghc-lib-parser 9.2.5.20221107
```

## Description
When using the `haddock-style: single-line` option, `fourmolu` transforms certain multi-line comments into single-line comments that result in compilation failures.

## Example
Suppose we have the following function definition:
```hs
{- | Accepts a unit, and produces a unit.

This function is pointless, in both senses of the word, as:

* It serves no useful purpose.
* Its definition has no point.

-}
foo :: () -> ()
foo = id
```

If we format this with `ormolu`, it produces the following:
```patch
-{- | Accepts a unit, and produces a unit.
-
-This function is pointless, in both senses of the word, as:
-
-* It serves no useful purpose.
-* Its definition has no point.
-
--}
+-- | Accepts a unit, and produces a unit.
+--
+-- This function is pointless, in both senses of the word, as:
+--
+-- * It serves no useful purpose.
+-- * Its definition has no point.
 foo :: () -> ()
 foo = id
```
So far so good: `ormolu` has inserted a space between each `--` and the start of each line within the comment.

But if we run `fourmolu` on the same function with the `haddock-style: single-line` option (thus mimicking the default configuration of `ormolu`, it produces the following:
```patch
-{- | Accepts a unit, and produces a unit.
-
-This function is pointless, in both senses of the word, as:
-
-* It serves no useful purpose.
-* Its definition has no point.
-
--}
+-- | Accepts a unit, and produces a unit.
+--
+--This function is pointless, in both senses of the word, as:
+--
+--* It serves no useful purpose.
+--* Its definition has no point.
 foo :: () -> ()
 foo = id
```
Note the lack of space between `--` and `*`.

This causes a compilation failure:
```hs
lib/Example.hs:12:1: error: parse error on input `--*'
   |
12 | --* It serves no useful purpose.
```

**Comment by brandonchinn178** [comment]
Are you on the newest version? Fourmolu 0.10.1.0 should have the fix for this

**Comment by jonathanknowles** [comment]
> Are you on the newest version? Fourmolu 0.10.1.0 should have the fix for this

@brandonchinn178 it's the version I got when I installed it this morning. 😄 

I'll update and try again.

**Comment by jonathanknowles** [comment]
> Yeah, I'm unable to repro on fourmolu 0.10.1.0. How did you install fourmolu?

I simply used:
```
$ cabal update
$ cabal install fourmolu
```
Which resulted in:
```
$ fourmolu --version
fourmolu 0.9.0.0 UNKNOWN UNKNOWN
using ghc-lib-parser 9.2.5.20221107
```

However, I suspect this is because my build environment was configured to use GHC 8.10.7. According to the `cabal` dependency resolver, the version of `base` distributed with GHC 8.10.7 is not compatible with the latest version of `fourmolu`.

I've just switched to GHC 9.2.5, and I'm now able to install `fourmolu` version `0.10.1.0`.

I'll try to reproduce it again and report back!
